spec/amber/controller/redirect_spec.cr

```diff
@@ -0,0 +1,112 @@
+require "../../../spec_helper"
+
+module Amber::Controller
+  describe Redirector do
+    describe "#redirect" do
+      it "redirects to location" do
+        controller = build_controller
+        redirector = Redirector.new("/some_path")
+
+        redirector.redirect(controller)
+
+        controller.response.headers["location"].should eq "/some_path"
+        controller.response.status_code.should eq 302
+        controller.context.content.nil?.should eq false
+      end
+
+      it "raises Exception::Controller::Redirect on invalid location" do
+        controller = build_controller
+
+        expect_raises Exceptions::Controller::Redirect do
+          redirector = Redirector.new("", params: {"user_id" => "123"})
+        end
+      end
+
+      context "with params" do
+        it "redirect to location and adds params to url" do
+          controller = build_controller
+          redirector = Redirector.new("/some_path", params: {"user_id" => "123"})
+
+          redirector.redirect(controller)
+
+          controller.response.headers["location"].should eq "/some_path?user_id=123"
+          controller.response.status_code.should eq 302
+          controller.context.content.nil?.should eq false
+        end
+      end
+
+      context "with flash" do
+        it "redirects to location and adds flash" do
+          controller = build_controller
+          redirector = Redirector.new("/some_path", flash: {"success" => "Record saved!"})
+
+          flash = controller.flash
+          redirector.redirect(controller)
+
+          controller.response.headers["location"].should eq "/some_path"
+          flash["success"].should eq "Record saved!"
+          controller.response.status_code.should eq 302
+          controller.context.content.nil?.should eq false
+        end
+      end
+
+      context "with a different status code" do
+        it "redirects to location and sets a status code" do
+          controller = build_controller
+          redirector = Redirector.new("/some_path", status = 301)
+
+          flash = controller.flash
+          redirector.redirect(controller)
+
+          controller.response.headers["location"].should eq "/some_path"
+          controller.response.status_code.should eq 301
+          controller.context.content.nil?.should eq false
+        end
+      end
+    end
+
+    describe ".from_controller_action" do
+      Spec.before_each do
+      end
+
+      it "redirects to correct location for given controller action" do
+        router = Amber::Router::Router.instance
+        router.draw :web do
+          get "/fake", HelloController, :index
+          get "/fake/:id", HelloController, :show
+          put "/hello/:id", HelloController, :update
+        end
+        controller = build_controller
+        redirector = Redirector.from_controller_action("hello", :show, params: {"id" => "11"})
```

Comment:
Add a test without params like `Redirector.from_controller_action("hello", :index)`, it looks like params are necessary looking at the method.